pandas/io/formats/format.py

```diff
@@ -1207,7 +1262,7 @@ def formatter(value):
 
         return formatter
 
-    def get_result_as_array(self):
+    def get_result_as_array(self) -> Union[ndarray, List[str]]:
```

Comment:
it depends on `self.fixed_width` L1303

`_trim_zeros_complex` L1688 returns List[str]

`_trim_zeros_float` L1706 returns List[str]

otherwise L1309 returns ndarray

Reply:
grr really? this just sounds wrong, can you cahnge those 2 to return an ndarray(object) if needed; these should all have a consistent signature.

Reply:
I really don't think we should be refactoring at the same time as adding type hints.

it's ok when someone is adding new code or modifying existing code.

but to benefit from the network effect of typing, i want to get this merged and get on with typing call sites and typing the imported functions.

refactoring can be done in parallel.

Reply:
ok sure